### PR TITLE
Support Unity Package Manager

### LINKED ISSUE #2338: Unity package for Unity SDK
**Is your feature request related to a problem? Please describe.**
When developing in Unity, I would like to be able to track my project's version of the Agones SDK. I would also like to be able to do this without manually downloading the code and placing it in my project's file structure.

**Describe the solution you'd like**
Unity has a package manager, which allows developers to add dependencies via a manifest file. https://docs.unity3d.com/Manual/upm-git.html It would be ideal if the Agones Unity SDK was configured to be imported into projects via this mechanism.

**Describe alternatives you've considered**
Currently, we just download the code and integrate it manually. If the UPM isn't supported, then we will continue to do this.

**Additional context**
Unity has its own package repository that this package could be added to, but IMO that would be overkill. Being able to add the package via a git url would definitely be sufficient.

### PR DESCRIPTION
**Pull request by aaronchapin-tocaboca**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation

/kind feature

> /kind hotfix

**What this PR does / Why we need it**:
Unity has its own package manager, which can be used to load external dependencies. This change allows unity developers to add the Agones SDK to their project by simply adding a line to their `manifest.json` file that follows this format:
 `"com.googleforgames.agones": "https://github.com/googleforgames/agones.git?path=/sdks/unity#<TAG>",`
Instead of manually downloading the SDK code locally and placing it in their project's folder structure.

**Which issue(s) this PR fixes**:
Closes #2338 

**Special notes for your reviewer**:
I took some educated guesses about the desired values for fields like the version number, author name/url etc. Any of these can be changed without an issue - just let me know what is preferred.

Added:
- Agones Unity SDK assembly definition file
- Package information json
- Meta files required by the Unity engine

### CODE REVIEW

File: sdks/unity/package.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+    "name": "com.googleforgames.agones",
+    "version": "1.18.1",
+    "displayName": "Agones Unity SDK",
+    "description": "The Unity Gameserver SDK for Agones",
+    "unity": "2019.1",
+    "author": {
+      "name": "Google For Games",
```
[the remark this answers is not in the document]
Comment (aaronchapin-tocaboca) [reply sdks/unity/package.json]: Updated 👍

File: sdks/unity/package.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+    "name": "com.googleforgames.agones",
+    "version": "1.18.1",
```
[the remark this answers is not in the document]
Comment (aaronchapin-tocaboca) [reply sdks/unity/package.json:3]: I answered this question in another thread, but just to have the info here as well: I tested this semantic versioning scheme and it is supported.

File: docs/governance/templates/release_issue.md [hunk 3]
```diff
@@ -80,6 +80,7 @@ and copy it into a release issue. Fill in relevant values, found inside {}
 - [ ] If full release move the [helm `Chart` version values][chart] is to {version}+1-dev
 - [ ] If full release, change to the `sdks/nodejs` directory and run the command `npm version {version}+1-dev` to update the package version
 - [ ] If full release move the [`sdks/csharp/sdk/AgonesSDK.nuspec` and `sdks/csharp/sdk/csharp-sdk.csproj`][csharp] to {version}+1-dev
+- [ ] If full release update the [`sdks/unity/package.json`][unity] package file's `Version` field to {version}+1-dev
```
Comment (markmandel) [inline docs/governance/templates/release_issue.md:83]: We should also add a checklist item in the `Update SDK Package Versions` section as well to increment the Unity version 👍🏻

Comment (aaronchapin-tocaboca) [reply docs/governance/templates/release_issue.md:84]: Done 👍

**Comment by markmandel** [comment]
Looks like a flake of `TestAutoscalerStressCreate`

**Review by markmandel — COMMENTED** [review]
This LGTM!

The only thing I just realised was outstanding on my end -

We should update https://agones.dev/site/docs/guides/client-sdks/unity/#download to have the new instructions on the best way to install.

We'll need to use the `feature` shortcode to hide and show the current vs new content:
https://agones.dev/site/docs/contribute/#within-a-page

**Review by markmandel — APPROVED** [review]

**Comment by markmandel** [comment]
Niced work!

**Pull request merged**